README.md

```diff
@@ -43,45 +46,67 @@ type Operation = {
 }
 ```
 
-The DID itself is derived from the sha256 hash of the first operation in the log. It is then base32 encoded and truncated to 24 chars.
+Each operation fully attests the current state of the document data. It also includes a content reference (hash) to the previous operation in the log, and is signed by a valid rotation key.
 
-To illustrate: 
-`did:plc:${base32Encode(sha256(createOp)).slice(0,24)}`
+The process for signing and hashing operation objects is to first encode them in the DAG-CBOR binary serialization format. [DAG-CBOR](https://ipld.io/specs/codecs/dag-cbor/spec/) is a restricted subset of the Concise Binary Object Representation (CBOR), an IETF standard (RFC 8949), with semantics and value types similar to JSON.
 
-Operations are verified, ordered and made available by the PLC server.
+For signatures, the DAG-CBOR bytes are signed, and then the signature bytes are encoded in to a string using `base64url` encoding.
 
-The operation logs are fully self-certifying construction of the current state of the DID document, with the exception of their ordering.
+For `prev` references, the SHA-256 of the previous operation's bytes are encoded as a "[CID](https://github.com/multiformats/cid)", using the relevant multibase code (for `dag-cbor`), and CIDv1 format.
+
+Rotation keys are serialized as strings using [did:key](https://w3c-ccg.github.io/did-method-key/), and only `secp256k1` ("k256") and NIST P-256 ("p256") are currently supported.
+
+The signing keys (`verificationMethods`) are represented as objects, with the actual keys in multibase encoding, as required by the DID Core specification.
```

Comment:
this is a bit unclear because they are similarly serialized as did:keys in the document data but represented as objects in the DID document presentation

Reply:
clarified